### PR TITLE
reduce cases of LogWarnings with InvalidHelix category in Pixel/Strip ClusterShape compatibility checks

### PR DESCRIPTION
**Pull request by slava77**
InvalidHelix is one of the leading LogWarnings in current pp processing.
It turns out that a large fraction (all?) of the warnings corresponds to the reference helix considered "invalid" because it was reduced to a straight line. This is actually a valid input.

FastCircle::isLine method is added to make the simple test.

There is no change to the physics results of the algorithms. A LogWarning is simply not issued anymore in the case that appears to already have a correct behavior.

### CODE REVIEW

File: RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc [hunk 1]
```diff
@@ -89,7 +89,9 @@ PixelClusterShapeSeedComparitor::compatible(const SeedingHitSet  &hits,
 { 
     if (!filterAtHelixStage_) return true;
 
-    if(!helix.isValid()) edm::LogWarning("InvalidHelix") << "PixelClusterShapeSeedComparitor helix is not valid, result is bad";
+    if(!helix.isValid() //check still if it's a straight line, which are OK
+       && !helix.circle().isLine())//complain if it's not even a straight line
+      edm::LogWarning("InvalidHelix") << "PixelClusterShapeSeedComparitor helix is not valid, result is bad";
```
Comment (slava77) [inline RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: @makortel @VinInn 
I wanted to check with you that I have correctly interpreted this case of using a "degraded" FastHelix.

Comment (VinInn) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: Interesting.... is the efficiency improving?

Comment (VinInn) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: Actually wonder what 
```float xc = helix.circle().x0(), yc = helix.circle().y0();```
would then do

Comment (slava77) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: there is no change in physics in this PR, it just reduces the range of cases when the logWarning is issued.

The xc,yc are (0,0) for the case of a line.

Comment (VinInn) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: ok, we need to have a further look on how to use a straight line

Comment (slava77) [reply RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeSeedComparitor.cc]: In the LogWarnings that I have investigated
this code is called by RecoTracker/TkSeedGenerator/plugins/SeedFromConsecutiveHitsTripletOnlyCreator.cc:53
there the code already apparently makes some assumptions about ```!helix.isValid()``` and already makes up a state at vertex for an invalid helix which simply points from the vertex to the outer hit (just as if it's a straight line)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/23209	3028a3f
- jenkins tests pass and comparisons with the baseline show changes in the log-warnings, indicating a reduction

**Comment by fabiocos** [comment]
+1

differences looks related to the known issue

**Pull request merged**